hugegraph-core/src/main/java/org/apache/hugegraph/backend/tx/EphemeralJobQueue.java

```diff
@@ -0,0 +1,236 @@
+        if (this.state.compareAndSet(State.INIT, State.EXECUTE)) {
+            try {
+                BatchEphemeralJob job = new BatchEphemeralJob(this);
+                EphemeralJobBuilder.of(this.graph.graph())
+                                   .name("batch-ephemeral-job")
+                                   .job(job)
+                                   .schedule();
+            } catch (Throwable e) {
+                // Maybe if it fails, consider clearing all the data in the pendingQueue,
+                // or start a scheduled retry task to retry until success.
+                LOG.warn("Failed to schedule RemoveLeftIndexJob", e);
+                this.pendingQueue.clear();
+                this.state.compareAndSet(State.EXECUTE, State.INIT);
+            }
+        }
+    }
+
+    public boolean isEmpty() {
+        return this.pendingQueue.isEmpty();
+    }
+
+    public static class BatchEphemeralJob extends EphemeralJob<Object> {
+
+        private static final int PAGE_SIZE = 100;
+        private static final String BATCH_EPHEMERAL_JOB = "batch-ephemeral-job";
+        private static final int MAX_CONSUME_COUNT = EphemeralJobQueue.CAPACITY / 2;
+
+        private WeakReference<EphemeralJobQueue> queueWeakReference;
+
+        public BatchEphemeralJob(EphemeralJobQueue queue) {
+            this.queueWeakReference = new WeakReference<>(queue);
+        }
+
+        @Override
+        public String type() {
+            return BATCH_EPHEMERAL_JOB;
+        }
+
+        @Override
+        public Object execute() throws Exception {
+            boolean stop = false;
+            Object ret = null;
+            int consumeCount = 0;
+            InterruptedException interruptedException = null;
+            EphemeralJobQueue queue;
+            List<EphemeralJob<?>> batchJobs = new ArrayList<>();
+            while (!stop) {
+                if (interruptedException == null && Thread.currentThread().isInterrupted()) {
+                    interruptedException = new InterruptedException();
+                }
+
+                queue = this.queueWeakReference.get();
+                if (queue == null) {
+                    stop = true;
+                    continue;
+                }
+
+                if (queue.isEmpty() || consumeCount > MAX_CONSUME_COUNT ||
+                    interruptedException != null) {
+                    queue.consumeComplete();
+                    stop = true;
+                    if (!queue.isEmpty()) {
+                        queue.reScheduleIfNeeded();
+                    }
+                    continue;
+                }
+
+                try {
+                    while (!queue.isEmpty() && batchJobs.size() < PAGE_SIZE) {
+                        EphemeralJob<?> job = queue.queue().poll();
+                        if (job == null) {
+                            continue;
+                        }
+                        batchJobs.add(job);
+                        consumeCount++;
```

Comment:
batchJobs.size() and consumeCount are the same?
we can remove the consumeCount if true, or else set consumeCount += batchJobs.size()
and prefer to do initJob() here for more readable.

Reply:
they are different,  batchJobs.size() is on one transaction execute, consumeCount is to count total execution count for  free up cpu when consumeCount exceed limit

Reply:
ok.
can we move `consumeCount++` to line 166 and use `consumeCount += batchJobs.size()` instead?

Reply:
there should count total job count more reasonable